File: src/main/java/com/itsci/mju/maebanjumpen/report/service/ReportService.java
package com.itsci.mju.maebanjumpen.report.service;

import com.itsci.mju.maebanjumpen.report.dto.ReportDTO;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

public interface ReportService {
    List<ReportDTO> getAllReports();
    @Transactional
        // @Transactional สำคัญมากสำหรับเมธอดนี้
    ReportDTO createReport(ReportDTO reportDto);
    ReportDTO getReportById(int id);
    void deleteReport(int id);
    List<ReportDTO> getReportsByStatus(String reportStatus);
    ReportDTO updateReport(int id, ReportDTO reportDto);
    Optional<ReportDTO> findByPenaltyId(Integer penaltyId);
    Optional<ReportDTO> findLatestReportWithPenaltyByPersonId(Integer personId); // ยังคงเหมือนเดิม
    Optional<ReportDTO> findByHireIdAndReporterId(Integer hireId, Integer reporterId);
    void updateUserAccountStatus(int personId, boolean isBanned);

}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/report/service/ReportService.kt
package com.itsci.mju.maebanjumpen.report.service

import com.itsci.mju.maebanjumpen.report.dto.ReportDTO
import com.itsci.mju.maebanjumpen.report.request.CreateReportRequest
import com.itsci.mju.maebanjumpen.report.request.UpdateReportRequest
import java.util.Optional

interface ReportService {
    fun listAllReports(): List<ReportDTO>

    fun createReport(request: CreateReportRequest): ReportDTO

    fun getReportById(reportId: Long): ReportDTO

    fun deleteReport(id: Long) : Boolean

    fun getReportsByStatus(reportStatus: String): List<ReportDTO>

    fun updateReport(request: UpdateReportRequest): ReportDTO

    fun findByPenaltyId(penaltyId: Long?): Optional<ReportDTO>

    fun findLatestReportWithPenaltyByPersonId(personId: Long?): Optional<ReportDTO>

//    fun findByHireIdAndReporterId(hireId: Long?, reporterId: Long?): Optional<ReportDTO>

    fun updateUserAccountStatus(personId: Long, isBanned: Boolean)
}

